src/tests/ut/medicine/medicine_test.hpp

```diff
@@ -0,0 +1,14 @@
+#include "medicine.hpp"
+#include "product.hpp"
+#include "gtest/gtest.h"
+
+struct MedicineTest : ::testing::Test
+{
+    const std::string name{"anaesthetic"};
+    constexpr int price{45};
+    constexpr int amount{100};
+    constexpr int expiration_date{10122025};
+    constexpr std::pair storage_temperature{-10, 0};
+    const std::vector<std::string> composition{"chemicals"};
+    anaesthetic = std::make_shared<Medicine>(name, price, amount, expiration_date, storage_temperature, composition);
+};
```

Comment:
eof , can be corrected in future 
we shall think about automating check for that